### DISCUSSION TITLE
Query number of Check-Clicks

### DISCUSSION THREAD

**Discussion by adlergfx** [LiaScript/LiaScript#147] [Q&A]
Hello, 
i am very happy with lia and try currently some Sandbox applications. Am I able to detect how many times a user has clicked "check"? I can disable it, but it would be great to understand if a user has used a bruteforce approach (check to win). AND can i figure out if hints would be used?

 If not using with classrooms i think to submit success to a local db, but for improvement of courses, it would be great to know if hints and checks would be used. It can be a hint if the lecture should be improved in that topic.

**Comment by andre-dietrich** [comment]
Dear @adlergfx ... you can simply attach a script to a quiz, that checks the input ... In LiaScript `@input` is always used as a placeholder for the Editor-, Quiz-, or Survey-Input. I added also a reusable macro, such that you do not have to attach the same script over and over to your quizzes. In LiaScript-Macros `@0...@9` are used as place-holders for the input parameters:

``` markdown
<!--

@onload
window.quizCounter = {}
@end

@quizCounter: @quizCounterHelper(@uid,`@0`)

@quizCounterHelper
<script>
    // storing the counts in a global variable
    if (!window.quizCounter['@0']) {
        window.quizCounter['@0'] = 0
    }

    window.quizCounter['@0']++

    window.console.log("the input was '@input'", "trial: ", window.quizCounter['@0'])

    window.console.log(window.quizCounter)

    '@input' == '@1'
</script>
@end

-->

# QuizCounts

You can attach a script-tag to your quiz, which basically control the behaviour or can send some information to a remote server. In the example below, the click is simply stored within a global variable.

what is $ 22 \cdot 22 $

[[484]]
<script>
    // storing the counts in a global variable
    if (window.counter == undefined) {
        window.counter = 0
    }
    window.counter ++

    window.console.log("the input was '@input'", "trial: ", window.counter)
    
    @input == 484
</script>

## Using the macro

With the `@quizCounter` you can make the upper script reusable. This macro will call another helper-macro, and pass in a `uid`, which is a special macro, that generates a unique id, whenever it is called. This way the result is stored within a global dictionary.

[[484]]
@quizCounter(484)


[[484]]
@quizCounter(111)

As an alternative you could also call the helper directly, but then you will have to pass a unique id by yourself...

[[111]]
@quizCounterHelper(MyID,123)

Macros are simple substitutions, so spaces are injected too ...
``` 

You can try this out at the LiveEditor: [here](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA61UTU8bMRC9+1cMH1KC2CyEcqgQQanaQzlwqNSqqmikeL2TrItjL/5I2CL+e8feDQ0l6aXdQ+L1PM88v3mzl3uDAWNjo5XhJVtJXZpVfh/kz/cmaI8WRvD4xMaoS0Jt7F/A5ttHVDXa/jjIMpuOT6dHL8FtmF06YWXtrxjQc3ICzhsr9Rx8hSAi0oHUwGGuTMEVLLmVvFCY4HIG/b3X9G5749Pe5AgeEyg+uzB0kdMEemLsb7jj4xdxYbQzCnNl5v39SFTqOnhYcQe9cVr39jPY90RVXQAtd3Lcmfb1iQ67LgCjEa2HPXZ5slawbchgcMXYAXxan3SMfTMBBCcVveeiIjHbEwPPSWcDjQkWYiUiWkkCFNxJwZVqqAPaW6NSNwqs+FJGrLEpnaN64Mwi3n9m7IJ7aXRMyMHiwngkhF2izeFapwz4wBe1ipmUWWVti5UUdyAdOEmhJrUfS1LEV9vanjO2qriPBw7h7Ay+i9L4uDhk7Pb2/O35ZPLPjnruRuf1EQRd4kxqLLd46hn27KQtwf/kH7F2QsyVflp05Eh337ACOziAL2596wUX1jD2lURN79PNOZzG/qd+LvgdpnioaTI7k1Arg0vSw+eKZE+5iJBSEC0CXBs6YqFK4zxI4Yx2S6i565Se0idgujYX5SAD1ijoctDBfezpHDVa7jHGg5b3gXQp4ynaJxeBTG2PRbHsyKx4kwhbdEGl8A7/lFJEc3Lb5L+NsqlCn7ZowrbHhsMhxd4RMUqpaEeT1ZfYCmeCKmnXmVaPSKfVgopaFF41GRTUIwrodCJpR6OEcVSSSBv3haJJ8+hQzfI8kaXqfxDqvqw3zfWHbHj2hrjdRB0pkcV2kOgvFM5LH+K1XUZjSppzgS1G6h/EjHTyxgCV+QX64hBY8QUAAA==)

![image](https://github.com/LiaScript/LiaScript/assets/3089101/7e9693ef-30fe-4fff-b394-21b44fcf2852)
---
I hope this can be used as a bootstrap for your purpose ...
